enterprise/server/byte_stream_server_proxy/byte_stream_server_proxy.go

```diff
@@ -298,6 +301,10 @@ func (s *ByteStreamServerProxy) writeLocalOnly(stream bspb.ByteStream_WriteServe
 func (s *ByteStreamServerProxy) dualWrite(ctx context.Context, stream bspb.ByteStream_WriteServer) error {
 	// Grab the first frame from the client so the local writer can be created
 	req, err := stream.Recv()
+	if err == io.EOF {
+		log.CtxWarningf(ctx, "Unexpected EOF reading first frame: %v", err)
```

Comment:
nit: maybe info-level since this is a client error?

```suggestion
		log.CtxInfof(ctx, "Unexpected EOF reading first frame: %v", err)
```